src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemProvider.cs

```diff
@@ -88,5 +84,34 @@ private bool GetSearchCurrentDocumentOptionWorker(INavigateToCallback callback)
             var options2 = callback.Options as INavigateToOptions2;
             return options2?.SearchCurrentDocument ?? false;
         }
+
+        public void StartSearch(INavigateToCallback callback, string searchValue, INavigateToFilterParameters filter)
+        {
+            StartSearch(callback, searchValue, filter.Kinds);
+        }
+
+        private void StartSearch(INavigateToCallback callback, string searchValue, ISet<string> kinds)
+        {
+            this.StopSearch();
```

Comment:
can we bail out *very quickly* if 'kinds' just contains File and nothing else we support?  i.e. intersect this with KindsProvided and do nothing if we get the empty set...

Reply:
It was my understanding that the environments wouldn't call this method if the filter wouldn't match any known kinds.

Reply:
Ah, i see then.  So we'll be queried for what we support.  And they won't bother if we don't support anything they want.  That makes sense then.  It might make sense to check/assert if that happens though as it would indicate a bug in the caller.

Reply:
That said, i think we should still optimize the case where we are queried for *everything* we support.  IN that case, filtering is just a tax as we'll always succeed.  And i don't want any extra cpu cycles in nav-to :)